packages/dashboard-core-plugins/src/GridWidgetPlugin.tsx

```diff
@@ -36,6 +40,36 @@ export function GridWidgetPlugin(
     };
   }, [dh, fetch]);
 
+  useEffect(
+    function startListeningModel() {
+      if (!model) {
+        return;
+      }
+
+      // If the table inside a widget is disconnected, then don't bother trying to listen to reconnect, just close it and show a message
+      // Widget closes the table already when it is disconnected, so no need to close it again
+      function handleDisconnect() {
+        setError(new Error('Model disconnected'));
```

Comment:
Just curious if we maybe want this to say `Table disconnected` which might be slightly more user friendly?